### DISCUSSION TITLE
"Today" in Timeframe instead of 24 hours or 1 Day

### DISCUSSION THREAD

**Discussion by adhyayan07** [jertel/elastalert2#604] [Q&A] [answered]
What i have to do is, make elastalert to scan from the start of the day to current time instead of whole 24 hours from previous day to current time.

E.g :

Current behaviour:
                  "At least 10 events occurred between 2021-12-09 **12:30 IST and 2021-12-10 12:30** IST"

Expected behaviour:
                  "At least 10 events occurred between 2021-12-10 **00:00 IST and 2021-12-10 12:30** IST"

Any way to achieve this?

**Comment by jertel** [comment, marked as answer]
ElastAlert 2 is an always-on monitoring process, so asking for a way to have it scan a specific time is a bit of a conflict with how it was designed. In a traditional configuration, it starts up and looks back X amount of time based on configuration/rule settings, determines if any rules need triggered, records where it left off (date/time) and then goes to sleep for a few minutes (default is 5). When it wakes up again, it scans backwards from the date/time where it left off before it slept, to the current time, and determines if any new data, combined with the data from the previous runs, has exceeded a rule trigger. This process repeats continually. Again, every time it wakes up it's only scanning for that short period when it was sleeping. This avoids heavy scans all day long against Elasticsearch.

So when you ask how to make it scan from the start of the day to the current time, I struggle to envision how this would work. Would it go to sleep for 5 minutes, wake up, and re-scan the data again from that same timeline + the 5 minutes that elapsed while it slept, and then near the end of the day it's scanning almost 24 hours of data every time it runs?

Based on this understanding, the closest parameter that I can think of that might be something worth looking into is the [scan_entire_timeframe](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#scan-entire-timeframe) rule configuration parameter.

Alternatively, if you're only wanting to run ElastAlert 2 manually or on an externally managed scheduler (like crontab), perhaps you could use the included [test_rule.py](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#testing-your-rule) tool, which allows you to specify exact start and end times. This tool has an option that allows you to enable the actual alerts, though by default it disabled since the tool is primarily intended for debugging to make sure your rule works.
